modules/core/src/main/java/org/apache/ignite/internal/processors/cluster/baseline/autoadjust/ChangeTopologyWatcher.java

```diff
@@ -99,17 +99,26 @@ public ChangeTopologyWatcher(GridKernalContext ctx) {
         if (discoEvt.eventNode().isClient() || discoEvt.eventNode().isDaemon())
             return;
 
-        synchronized (this) {
-            lastBaselineData = lastBaselineData.next(discoEvt.topologyVersion());
+        triggerBaselineUpdate(discoEvt.topologyVersion());
+    }
 
-            if (isLocalNodeCoordinator(discoveryMgr)) {
-                exchangeManager.affinityReadyFuture(new AffinityTopologyVersion(discoEvt.topologyVersion()))
+    /**
+     * Schedule update of the baseline topology
+     * @param topologyVersion version of topology
+     */
+    public synchronized void triggerBaselineUpdate(long topologyVersion) {
```

Comment:
I believe it is better not to add second method here. You can rename onEvent -> triggerBaselineUpdate and class ChangeTopologyWatcher to something like BaselineTopologyUpdater.
And then registration of events(EVT_NODE_FAILED, EVT_NODE_LEFT, EVT_NODE_JOINED) should be modified according to using triggerBaselineUpdate(long topologyVersion)